controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java

```diff
@@ -29,10 +31,12 @@
 
     private final ZoneRegistry zoneRegistry;
     private final AthenzClientFactory athenzClientFactory;
+    private final ApplicationPackage applicationPackage;
 
-    public DeployAuthorizer(ZoneRegistry zoneRegistry, AthenzClientFactory athenzClientFactory) {
+    public DeployAuthorizer(ZoneRegistry zoneRegistry, AthenzClientFactory athenzClientFactory, ApplicationPackage applicationPackage) {
```

Comment:
Why is `ApplicationPackage` a constructor argument. I would make it an argument to `throwIfUnauthorizedForDeploy` instead.